hawkular-inventory-impl-cassandra/src/main/java/org/hawkular/inventory/impl/cassandra/CassandraBackend.java

```diff
@@ -159,8 +170,59 @@ public CassandraBackend(RxSession session) {
     @Override
     public Iterator<Row> getTransitiveClosureOver(Row startingPoint, Relationships.Direction direction,
                                                        String... relationshipNames) {
-        //TODO implement
-        return null;
+        String startingCP = startingPoint.getString(Statements.CP);
+        if (startingCP == null) {
+            // Not a vertex
+            return Collections.<Row>emptyList().iterator();
+        }
+        Observable<Row> closureContent = getRelationshipTables(direction)
+                .flatMap(table -> {
+                    Statement q = select().all().from(table).where(in(Statements.NAME, Arrays.asList(relationshipNames)));
+                    return session.execute(q);
+                })
+                .map(relationships -> computeTransitiveClosure(startingCP, relationships))
+                .flatMap(this::loadEntities);
+        return closureContent.toBlocking().getIterator();
+    }
+
+    private Observable<String> getRelationshipTables(Relationships.Direction direction) {
```

Comment:
We need to create an observable pipeline soon or later, and it's a pretty way (well, "pretty" is subjective) to handle single table case (incoming and outgoing) and multiple tables case (both) in the same, general way. So there's no specific code for "both".